### PR TITLE
Remove global automation from MidiImport

### PR DESCRIPTION
**Pull request by Veratil**
Completes step 1 from #403

### CODE REVIEW

File: plugins/MidiImport/MidiImport.cpp [hunk 1]
```diff
@@ -323,19 +333,12 @@ bool MidiImport::readSMF( TrackContainer* tc )
...
-			printf("Another timesig at %f\n", timeSig.beat);
-			timeSigNumeratorPat->putValue( timeSig.beat*ticksPerBeat, timeSig.num );
-			timeSigDenominatorPat->putValue( timeSig.beat*ticksPerBeat, timeSig.den );
-		}
-		else
-		{
-		}
-
+		timeSigNumeratorPat->putValue(timeSig.beat*ticksPerBeat, timeSig.num);
+		timeSigDenominatorPat->putValue(timeSig.beat*ticksPerBeat, timeSig.den);
```
Comment (PhysSong) [inline plugins/MidiImport/MidiImport.cpp]: From https://github.com/LMMS/lmms/wiki/Coding-Conventions#infix-operators:
> Spaces MUST be before and after infix operators (`=`, `+`, `-`, `*`, `/`, etc.).

**Comment by PhysSong** [comment]
The rest of this PR looks fine.

File: plugins/MidiImport/MidiImport.cpp (hunk 1 above)
Comment (Veratil) [reply plugins/MidiImport/MidiImport.cpp]: Oops! I removed the spaces around `()` but forgot that. 😄

**Comment by Reflexe** [comment]
This PR is broken with VS builds. see https://ci.appveyor.com/project/Lukas-W/lmms/builds/28005325/job/1d524yxraw2rrw5q

**Comment by Veratil** [comment]
> 
> 
> This PR is broken with VS builds. see https://ci.appveyor.com/project/Lukas-W/lmms/builds/28005325/job/1d524yxraw2rrw5q
> 
> ```
> MidiImport.obj : error LNK2001: unresolved external symbol "public: virtual void __thiscall TrackContentObject::movePosition(class MidiTime const &)" (?movePosition@TrackContentObject@@UAEXABVMidiTime@@@Z) [C:\projects\lmms\build\plugins\MidiImport\midiimport.vcxproj]
> MidiImport.obj : error LNK2001: unresolved external symbol "public: virtual void __thiscall TrackContentObject::changeLength(class MidiTime const &)" (?changeLength@TrackContentObject@@UAEXABVMidiTime@@@Z) [C:\projects\lmms\build\plugins\MidiImport\midiimport.vcxproj]
> C:\projects\lmms\build\plugins\Debug\midiimport.dll : fatal error LNK1120: 2 unresolved externals 
> ```

I don't see how this PR would affect that, I didn't touch those functions.

**Comment by PhysSong** [comment]
I'm not sure why MinGW builds pass while MSVC builds fail, but I found the reason: class `TrackContentObject` is not marked as exported. Changing `class TrackContentObject` to `class LMMS_EXPORT TrackContentObject` in `Track.h` will fix this.

**Comment by PhysSong** [comment]
I found that MIDI import still uses global automation because `Song::tempoAutomationPattern()` uses it.

**Comment by Veratil** [comment]
> 
> 
> I found that MIDI import still uses global automation because `Song::tempoAutomationPattern()` uses it.

There's also a few calls to `AutomationPattern::globalAutomationPattern` which in turn calls `Engine::getSong()->globalAutomationTrack()`. The hole gets deeper. 🤔

**Comment by PhysSong** [comment]
> There's also a few calls to `AutomationPattern::globalAutomationPattern` which in turn calls `Engine::getSong()->globalAutomationTrack()`.

True, but after removing `Song::tempoAutomationPattern` and its usage, new global automation patterns can be created only by manual GUI actions.

**Pull request merged**